src/resource.ts

```diff
@@ -40,13 +45,13 @@ const MUTATING_ACTIONS: [Action, Action, Action] = [
     Action.Delete,
 ];
 
-export type HandlerSignature<T extends BaseModel> = Callable<
-    [Optional<SessionProxy>, any, Dict, LoggerProxy],
+export type HandlerSignature<T extends BaseModel, TC extends BaseModel> = Callable<
+    [Optional<SessionProxy>, any, Dict, TC, LoggerProxy],
```

Comment:
I know it's a generic type param but could you rename `TC` to something a bit more readable (e.g. `TypeConfiguration`)?